src/lib/annotations/doc/DocHighlightDialog.js

```diff
@@ -293,9 +293,12 @@ class DocHighlightDialog extends AnnotationDialog {
             ]);
             annotatorUtil.showElement(highlightLabelEl);
 
-            // Hide delete button on plain highlights if user doesn't have
-            // permissions
-            if (annotations[0].permissions && !annotations[0].permissions.can_delete) {
+            if (!this.canAnnotate) {
```

Comment:
Hides buttons in the plain highlight dialog if the user does not have `can_annotate` permissions